servers/audio/effects/audio_effect_pitch_shift.cpp

```diff
@@ -160,8 +160,10 @@ void SMBPitchShift::PitchShift(float pitchShift, long numSampsToProcess, long ff
 
 			/* ***************** PROCESSING ******************* */
 			/* this does the actual pitch shifting */
-			memset(gSynMagn, 0, fftFrameSize*sizeof(float));
-			memset(gSynFreq, 0, fftFrameSize*sizeof(float));
+			size_t fftBufferSize = static_cast<size_t>(fftFrameSize) * sizeof(float);
+			ERR_FAIL_COND(fftBufferSize > MAX_FRAME_LENGTH);
```

Comment:
If this function is called a lot, it might be worth saving the error spam just in case the compiler is right and it does hit this fail condition:

```
if (fftBufferSize > MAX_FRAME_LENGTH)
{
ERR_PRINT_ONCE(...);
return;
}
```
Just logging an error could go horribly wrong actually if this is in an audio callback, but hopefully it will give some indication this is hitting, and better than running with a buffer overrun.

Reply:
I made that change, also using `unlikely`.